### DISCUSSION TITLE
How to have color opacity with custom colors?

### DISCUSSION THREAD

**Discussion by smhmd** [tailwindlabs/tailwindcss#2051] [Help]
I have custom colors `theme: {extend: {colors: {...}}}`. How to construct them in a way to be able to use things like `placeholder-opacity-25` and so on?

**Comment by estevanmaito** [comment]
Are you having any problem with this?

Because if you create a color the way you suggested, it will be available everywhere, eg (extending with a color named 'primary'): `bg-primary`, `text-primary`, `placeholder-primary`, etc. And other utilities would be correctly applied to it.

Just remind that for `placeholder-opacity-*` to work, it must be in the same element as the placeholder color, see https://tailwindcss.com/docs/placeholder-opacity/#usage

**Reply by smhmd** [reply]
Sorry for leaving it out (thought it was too much detail) but my colors are css variables:
```js
  theme: {
    extend: {
      colors: {
        theme: {
          primary: 'var(--primary)',
          secondary: 'var(--secondary)',
          ternary: 'var(--ternary)',
          disabled: 'var(--disabled)',
        },
      },
    },
  },
```

**Reply by hacknug** [reply]
Then you should probably use a similar approach to the one I use here: https://codepen.io/hack_nug/pen/VwLPBzw

**Reply by smhmd** [reply]
@hacknug To be honest, writing the occasionally needed class in CSS (e.g., `placeholder-theme-secondary placeholder-opacity-25`) sounds preferable to recreating Tailwind in CSS -- unless I understood what's in the codepen wrong. I should mention that I use these custom properties for dark mode.

**Reply by hacknug** [reply]
Of course I wasn't suggesting you to manually recreate Tailwind in CSS. I shared an example showing you how to use custom properties in Tailwind-like classes. Those values you see there, you can use them on your `tailwind.config.js`.

**Comment by iaminamcom** [comment]
I've same need. @adamwathan can you help us.

**Comment by Waseem-Almoliky** [comment]
This Post here seems interesting.
have a look.
although it needs to make tailwind `important` property on witch is some thing I don't prefer.
[https://github.com/tailwindlabs/tailwindcss/issues/1692#issuecomment-762731385](https://github.com/tailwindlabs/tailwindcss/issues/1692#issuecomment-762731385)
